Getting Started Guide (#96)
NCAR/geocat-applications: GETTING_STARTED.md

# Getting Started

## What is GeoCAT Applications?

```{include} README.md
:language: md
:start-after: <!-- readme_start -->
:end-before: <!-- readme_end -->
```

## New to Python?
If you are new to Python or want to learn more, [Project Pythia](https://projectpythia.org/) offers high-quality,
geoscience-oriented Python tutorials

[Pythia Foundations](https://foundations.projectpythia.org/landing-page.html) covers prerequisites that
will be helpful to learn for GeoCAT Applications like:

- Getting Started with Python
- Getting Started with Jupyter
- Working with NumPy and Pandas
- Plotting with Matplotlib

## New to Jupyter Notebooks?

[Jupyter Notebooks](https://docs.jupyter.org/en/latest/what_is_jupyter.html#what-is-a-computational-notebook-anyway)
provide an interactive interface to write, execute, and visualize code. Traditionally, code is written in a script and

> executed in order and completely run from beginning to end each time a script is run. Instead, Jupyter Notebooks separate
> a script into indvidiual cells that can be executed individually and in any order. Each cell in an IPython Jupyter notebook
> can be filled with Python code, markdown, or raw code. Jupyter notebooks has grown in popularity because it offers an simple
> method to include plain english description along with a coding script to improve readability. In addition, because each cell
> can be run in any order, it is possible to organize a notebook to avoid having to constantly re-run parts of a script that might
> take longer to run than others, like like retrieving data. The outputs of each cell appears below the cell without the
> need for typing `print` or `plt.show()`.

Reviewer: ```suggestion
executed in order from beginning to end each time a script is run. Instead, Jupyter Notebooks allow you to divide code into cells that can be executed individually and in any order. Each cell in a Jupyter Notebook
can be filled with executable code, markdown, or raw code. Jupyter Notebooks have grown in popularity because they offer a simple
method to include plain English description alongside code to improve readability. In addition, because each cell
can be run in any order, it is possible to organize a notebook to avoid having to constantly re-run parts of a script that might
take longer to run than others, like like retrieving data. The outputs of each cell appear below the cell without the
need for typing `print` or `plt.show()`.
```

A few edits for clarity / typos.

Re: IPython, I think we should stick to Jupyter since the rebranding from IPython Notebooks was a while back now and without context it might be more confusing than helpful to reference both.

# Getting Started

## What is `geocat-applications`?

> GeoCAT Applications is a community resource managed by the GeoCAT team. Inspired by

Author: I'll look into it, I've seen the ability to import markdown files in other programs, not sure about here though

> GeoCAT Applications is a community resource managed by the GeoCAT team. Inspired by

Author: So by default, I don't believe Markdown has this capability. There are some outside packages that could make this work (see: [list of packages](https://stackoverflow.com/questions/18438907/embedding-one-markdown-document-in-another), [mdx](https://github.com/mdx-js/mdx), [pandoc-include](https://github.com/DCsunset/pandoc-include)) but I think they might be more trouble than they are worth for this short section

> GeoCAT Applications is a community resource managed by the GeoCAT team. Inspired by

Author: I see the sphinx option, I'll see about adding it in

[NCL Applications](https://www.ncl.ucar.edu/Applications/).

GeoCAT Applications is broken into two main sections:
1. [Applications](https://ncar.github.io/geocat-applications/)
2. [NCL to Python](https://ncar.github.io/geocat-applications/ncl/ncl_index/ncl_index.html)

### Applications
Applications is designed to be a quick reference to demonstrate capabilities within the scientific
Python ecosystem that may be relevant to your geoscience workflows

### NCL to Python

> NCL to Python consists of pages that explain specifically how to achieve something that was possible

Reviewer 2: ```suggestion
NCL to Python consists of pages that explain specifically how to achieve something that is possible
```

in NCL in Python, including any algorithmic differences, guidance regarding replication under different

> conditions or circumstances, and any other relevant comparisons between the NCL and Python functionality

Reviewer 2: ```suggestion
conditions or circumstances, and any other relevant comparisons between the NCL and Python functionality.
```

> This is part of the GeoCAT's commitment to NCL and NSF NCAR's [Pivot to Python Initiative](https://www.ncl.ucar.edu/Document/Pivot_to_Python/)

Reviewer 2: ```suggestion
This is part of the GeoCAT team's commitment to NCL and NSF NCAR's [Pivot to Python Initiative](https://www.ncl.ucar.edu/Document/Pivot_to_Python/)
```

> These pages assume that the user has a working knowledge of NCL and are looking for transitional

Reviewer 2: ```suggestion
These pages assume that the user has a working knowledge of NCL and is looking for transitional
```

resources for specific functions. They also are not intended to be a comprehensive explanation of
the Python recommendations, but rather a guide for users who are already familiar with the NCL

> function and are looking for “equivalent” Python code

Reviewer 2: ```suggestion
function and are looking for “equivalent” Python code.
```

## New to Python?
If you are new to Python or want to learn more, [Project Pythia](https://projectpythia.org/) offers high-quality, geoscience-oriented
Python tutorials

[Pythia Foundations Book](https://foundations.projectpythia.org/landing-page.html) covers prerequisites that
will be helpful to learn for `geocat-applications` like:

- Getting Started with Python
- Getting Started with Jupyter
- Working with NumPy and Pandas
- Plotting with Matplotlib

Each tutorial and resource page of `geocat-applications` is built from [Jupyter notebooks](https://jupyterbook.org/en/stable/intro.html).

> Jupyter splits a Python script into individual "cells" to run separately. On the `geocat-application` webpage

Reviewer 2: ```suggestion
Jupyter splits a Python script into individual "cells" to run separately. On the `geocat-application` webpage,
```

> each page is a static resource, but `geocat-applications` can installed locally and run as a Jupyter notebook

Reviewer 2: ```suggestion
each page is a static resource, but `geocat-applications` can be installed locally and run as a Jupyter notebook.
```

Python scripts are executed in order and completely run from beginning to end each time a script is run. However,

> Jupter notebooks allow each cells to be executed individually and in any order. Jupyter notebooks hold snippets of

Reviewer 2: ```suggestion
Jupter notebooks allow each cell to be executed individually and in any order. Jupyter notebooks hold snippets of
```

Python code in "cells" and can be surrounded with additional cells with descriptions and images. This can be useful
if one cell contains Python code that takes a long time to run, like retrieving data. The outputs of each cell appears

> below each cell without the specific need for to `print` or `plt.show()`

Reviewer 2: ```suggestion
below each cell without the specific need for typing `print` or `plt.show()`.
```

> below each cell without the specific need for typing `print` or `plt.show()`.

Reviewer: This feels like a section on its own.  

I think we could also link out a bit more to Project Jupyter resources (e.g. demos [here](https://jupyter.org/try) or [here](https://docs.jupyter.org/en/latest/start/index.html), [docs about interactive programming / the REPL](https://docs.jupyter.org/en/latest/what_is_jupyter.html#interactive-programming-the-repl), or overviews such as [What is Jupyter?](https://docs.jupyter.org/en/latest/what_is_jupyter.html) or [What is a Notebook?](https://docs.jupyter.org/en/latest/#what-is-a-notebook)).

> If you have any questions, you can also reach us by email at [geocat@ucar.edu](geocat@ucar.edu)

Reviewer 2: ```suggestion
If you have any questions, you can also reach us by email at [geocat@ucar.edu](geocat@ucar.edu).
```

> If you have any questions, you can also reach us by email at [geocat@ucar.edu](geocat@ucar.edu).

Author: Link to the [Pythia](https://projectpythia.org/) support resources?

> If you have any questions, you can also reach us by email at [geocat@ucar.edu](geocat@ucar.edu).

Reviewer: Ah, sorry that wasn't clear.  I meant the "Getting Support" section of the sidebar on GeoCAT Applications with a link to GitHub Issues and the feature request form, but it's less robust than I thought and I'm not sure we can really link there the way it's set up. 

Maybe we could add a link to [GitHub Discussions](https://github.com/NCAR/geocat-applications/discussions) and/or [GitHub Issues](https://github.com/NCAR/geocat-applications/issues) as well as the email for now?  I think pushing conversations to GitHub as much as possible is probably a good idea (it's public for others to reference and shows activity on the repo).
